homeassistant/components/snapcast/config_flow.py

```diff
@@ -0,0 +1,77 @@
+"""Snapcast config flow."""
+
+from __future__ import annotations
+
+import logging
+import socket
+
+import snapcast.control
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DEFAULT_PORT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SNAPCAST_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
+    }
+)
+
+
+class SnapcastConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Snapcast config flow."""
+
+    async def async_step_user(self, user_input=None) -> FlowResult:
+        """Handle first step."""
+
+        def _show_form(errors=None):
+            return self.async_show_form(
+                step_id="user",
+                data_schema=SNAPCAST_SCHEMA,
+                errors=errors,
+            )
+
+        if not user_input:
+            return _show_form()
+
+        host = user_input[CONF_HOST]
+        port = user_input[CONF_PORT]
+        errors = {}
+        client = None
+
+        # Attempt to create the server - make sure it's going to work
+        try:
+            client = await snapcast.control.create_server(
+                self.hass.loop, host, port, reconnect=True
+            )
+        except socket.gaierror:
+            errors["base"] = "unknown"
+        except ConnectionRefusedError:
+            errors["base"] = "cannot_connect"
+        finally:
+            del client
+
+        await self.async_set_unique_id("Snapcast")
+        self._abort_if_unique_id_configured()
```

Comment:
This is a good point, but I don't really know how to solve it. The combination of host and port is unique because it is used to connect the server, but I think home-assistant does not consider that as a valid unique-id, right? The server API does not provide any identifier and even the mac is not unique because multiple instances could run on the same machine. Entities (clients and groups) still have valid ids.
So should I remove it entirely or is host+port ok?

Reply:
Removed it for now. Could be added later when a discovery flow is implemented.